security/medusa/l4-constable/chardev.c

```diff
@@ -469,36 +468,36 @@ inline static void ls_unlock(lightswitch_t* ls, struct semaphore* sem) {
  * eating one processor by a constable... ;) One can imagine
  * the performance improvement, and buy one more CPU in advance :)
  */
-static medusa_answer_t l4_decide(struct medusa_event_s * event,
+static authserver_answer_t l4_decide(struct medusa_event_s * event,
 		struct medusa_kobject_s * o1, struct medusa_kobject_s * o2)
 {
-	int retval;
+	authserver_answer_t retval;
 	teleport_insn_t tele_mem_decide[6];
 	struct tele_item *local_tele_item;
 	struct waitlist_item local_waitlist_item;
 
 	if (in_interrupt()) {
 		/* houston, we have a problem! */
 		med_pr_err("decide called from interrupt context :(\n");
-		return MED_ERR;
+		return AUTHS_NOT_REACHED;
 	}
 	if (am_i_constable() || current == gdb) {
```

Comment:
should be divided to two tests:
1) am_i_constable --> return AUTHS_ALLOW
2) current == gdb --> return AUTHS_DBG_ALLOW